### PR TITLE
Fixes #18313: implement nim package manager 

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/18313

### CODE REVIEW

File: tree/30_generic_methods/package_absent.cf [hunk 1]
```diff
@@ -24,9 +24,9 @@
 # @parameter_constraint version "allow_empty_string" : true
 # @parameter            architecture Architecture of the package, can be an architecture name  or "default" (defaults to "default")
 # @parameter_constraint architecture "allow_empty_string" : true
-# @parameter            provider Package provider to use, can be "yum", "apt", "zypper", "zypper_pattern", "slackpkg", "pkg", "ips" or "default" for system default package manager (defaults to "default")
+# @parameter            provider Package provider to use, can be "yum", "apt", "zypper", "zypper_pattern", "slackpkg", "pkg", "ips", "nimclient" or "default" for system default package manager (defaults to "default")
 # @parameter_constraint provider "allow_empty_string" : true
-# @parameter_constraint provider "select" : [ "", "default", "yum", "apt", "zypper", "zypper_pattern", "slackpkg", "pkg", "ips" ]
+# @parameter_constraint provider "select" : [ "", "default", "yum", "apt", "zypper", "zypper_pattern", "slackpkg", "pkg", "ips", "nimclient" ]
```
Comment (peckpeck) [inline tree/30_generic_methods/package_absent.cf:29]: Why nimclient and not just nim ?

**Review by peckpeck — CHANGES_REQUESTED** [review]
I don't see any package module file

File: tree/30_generic_methods/package_absent.cf (hunk 1 above)
Comment (ncharles) [reply tree/30_generic_methods/package_absent.cf:29]: because it a nim client and not a nim server

**Comment by ncharles** [comment]
package module file is already present in ncf: https://github.com/Normation/ncf/blob/master/tree/10_ncf_internals/modules/packages/nimclient

File: tree/30_generic_methods/package_absent.cf (hunk 1 above)
Comment (peckpeck) [reply tree/30_generic_methods/package_absent.cf:29]: it is a package type, how can a package type be a client or a server ?

Comment (ncharles) [reply tree/30_generic_methods/package_absent.cf:29]: If you use nim on AIX, you have a server that shares object, and a client that installs it, and the command is literraly "nimclient"

File: tree/20_cfe_basics/packages.cf [hunk 2]
```diff
@@ -119,11 +119,12 @@ bundle agent ncf_package(name, version, architecture, provider, state, options)
       "use_pkg_provider" expression => strcmp("pkg", "${provider}");
       "use_slackpkg_provider" expression => strcmp("slackpkg", "${provider}");
       "use_ips_provider" expression => strcmp("ips", "${provider}");
+      "use_nimclient_provider" expression => strcmp("nimclient", "${provider}");
       # Used in tests
       "use_test_provider" expression => strcmp("test", "${provider}");
 
     any::
-      "supported_provider" expression => "use_apt_provider|use_yum_provider|use_zypper_provider|use_pkg_provider|use_slackpkg_provider|use_test_provider|use_zypper_pattern_provider|use_ips_provider";
+      "supported_provider" expression => "use_apt_provider|use_yum_provider|use_zypper_provider|use_pkg_provider|use_slackpkg_provider|use_test_provider|use_zypper_pattern_provider|use_ips_provider|use_nim_client";
```
Comment (amousset) [inline tree/20_cfe_basics/packages.cf:127]:
```suggestion
      "supported_provider" expression => "use_apt_provider|use_yum_provider|use_zypper_provider|use_pkg_provider|use_slackpkg_provider|use_test_provider|use_zypper_pattern_provider|use_ips_provider|use_nimclient_provider";
```

**Review by peckpeck — APPROVED** [review]

**Pull request merged**